### PR TITLE
Mlr 158 - Authorization for Notification Service

### PR DESCRIPTION
**Pull request by mbucknell**
In addition to adding the authentication to the /notification endpoint I also refactored what was EmailTest into tests for the model (EmailTest) and tests for the EmailNotificationHandler service.

### CODE REVIEW

File: src/test/java/gov/usgs/wma/mlrnotification/EmailNotificationHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@ActiveProfiles("test")
+public class EmailNotificationHandlerTest {
+	
+	@Autowired
+	private EmailNotificationHandler emailHandler;
+
+private GreenMail testSmtp;
```
Comment (dsteinich) [inline src/test/java/gov/usgs/wma/mlrnotification/EmailNotificationHandlerTest.java]: This should be indented

File: src/test/resources/application-test.yml [hunk 2]
```diff
@@ -5,4 +5,18 @@ spring:
...
+  oauth2:
+    resource:
+      filter-order: 3
+      id: trusted-app
+      jwt:
+        keyUri: http://fake.gov/oauth/token_key
+  require-ssl: false
+
+server:
+    port: 8080
```
Comment (dsteinich) [inline src/test/resources/application-test.yml]: I'm not seeing why the server port need to be defined here. Can you tell me?

Comment (mbucknell) [reply src/test/resources/application-test.yml]: It probably doesn't. I was having some issues with running the tests when I did not have the profiles set correctly and wanted to make sure there was no conflict. I will take it out if it does not matter.

Comment (dsteinich) [reply src/test/resources/application-test.yml]: If a port is really needed in automated testing, the "webEnvironment=SpringBootTest.WebEnvironment.RANDOM_PORT" (see mlrLegacy's tests) is a better solution.

**Pull request merged**